src/pages/Explore/TraceExplorationPage.tsx

```diff
@@ -33,3 +43,30 @@ export function TraceExplorationView({ exploration }: { exploration: TraceExplor
     </UrlSyncContextProvider>
   );
 }
+
+const AdHocVariableFilterSchema = z.object({
+  key: z.string(),
+  operator: z.string(),
+  value: z.string(),
+});
+
+const InitialFiltersSchema = z.object({
+  filters: z.array(AdHocVariableFilterSchema),
+});
+
+/** Because the context comes from a different app plugin we cannot really count on it being the correct type even if
+ * it was typed, so it is safer to do runtime parsing here. It also can come from different app extensions and at this
```

Comment:
Yes, it's not ideal and it's also just temporary. The reason for it is that otherwise you have to bound `openApp` to current app plugin which can only been done by propagating it through some function args or props, but that then means this would already change existing API which means less space for exploration and changes later on.